components/sup/src/manager/service/spec.rs

```diff
@@ -71,6 +75,23 @@ impl FromStr for DesiredState {
     }
 }
 
+/// Helper enum to abstract over spec type.
+pub enum Spec {
+    Service(ServiceSpec),
+    Composite(CompositeSpec, Vec<ServiceSpec>),
+}
+
+impl Spec {
+    /// We need to get at the identifier of a spec, regardless of
+    /// which kind it is.
```

Comment:
I think you can safely lose this comment. All that information is communicated by the function signature